File: admin-frontend/src/schemas/permission.ts
import { CreatedUpdatedAt, PaginationParameters, UUIDSchema } from './generics';

export interface PermissionCreate {
  name: string;
  codename: string;
}

export interface PermissionUpdate {
  name?: string
  codename?: string;
}

interface BasePermission extends UUIDSchema, CreatedUpdatedAt {
  name: string;
  codename: string;
}

export interface Permission extends BasePermission { }

export interface PermissionEmbedded extends BasePermission { }

export interface PermissionListParameters extends PaginationParameters {
  query?: string;
}

File: fief/schemas/permission.py
from pydantic import BaseModel

from fief.schemas.generics import CreatedUpdatedAt, UUIDSchema


class PermissionCreate(BaseModel):
    name: str
    codename: str


class PermissionUpdate(BaseModel):
    name: str | None = None
    codename: str | None = None


class BasePermission(UUIDSchema, CreatedUpdatedAt):
    name: str
    codename: str


class Permission(BasePermission):
    pass


class PermissionEmbedded(BasePermission):
    pass
